pretyping/tacred.ml

```diff
@@ -62,30 +52,38 @@ let () = CErrors.register_handler (function
 let error_not_evaluable r = raise (NotEvaluableRef r)
 
 let is_evaluable_const env cst =
-  is_transparent env (ConstKey cst) && evaluable_constant cst env
+  is_transparent env (Evaluable.EvalConstRef cst) && evaluable_constant cst env
 
 let is_evaluable_var env id =
-  is_transparent env (VarKey id) && evaluable_named id env
+  is_transparent env (Evaluable.EvalVarRef id) && evaluable_named id env
+
+let is_evaluable_projection env p =
+  is_transparent env (Evaluable.EvalProjectionRef p)
 
 let is_evaluable env = function
-  | EvalConstRef cst -> is_evaluable_const env cst
-  | EvalVarRef id -> is_evaluable_var env id
+  | Evaluable.EvalConstRef cst -> is_evaluable_const env cst
+  | Evaluable.EvalVarRef id -> is_evaluable_var env id
+  | Evaluable.EvalProjectionRef p -> is_evaluable_projection env p
 
 let value_of_evaluable_ref env evref u =
   match evref with
-  | EvalConstRef con ->
+  | Evaluable.EvalConstRef con ->
     let u = Unsafe.to_instance u in
     EConstr.of_constr (constant_value_in env (con, u))
-  | EvalVarRef id -> env |> lookup_named id |> NamedDecl.get_value |> Option.get
-
-let evaluable_of_global_reference env = function
-  | GlobRef.ConstRef cst when is_evaluable_const env cst -> EvalConstRef cst
-  | GlobRef.VarRef id when is_evaluable_var env id -> EvalVarRef id
+  | Evaluable.EvalVarRef id ->
+    env |> lookup_named id |> NamedDecl.get_value |> Option.get
+  | Evaluable.EvalProjectionRef _ ->
+    assert false (* TODO *)
+
+let evaluable_of_global_reference env = function (* TODO projections *)
+  | GlobRef.ConstRef cst when is_evaluable_const env cst -> Evaluable.EvalConstRef cst
+  | GlobRef.VarRef id when is_evaluable_var env id -> Evaluable.EvalVarRef id
   | r -> error_not_evaluable r
 
 let global_of_evaluable_reference = function
-  | EvalConstRef cst -> GlobRef.ConstRef cst
-  | EvalVarRef id -> GlobRef.VarRef id
+  | Evaluable.EvalConstRef cst -> GlobRef.ConstRef cst
+  | Evaluable.EvalVarRef id -> GlobRef.VarRef id
+  | Evaluable.EvalProjectionRef _ -> assert false (* TODO *)
```

Comment:
Here is one place where it would make sense to have a projection constructor in `GlobRef.t`. Not sure what to do.

Reply:
Is this just a case where I should go through the compatibility constant for now? (And similarly for the other direction: detect that a constant is a compatibility constant, and create an `EvalProjectonRef`?)